packages/api/models/pdrs.js

```diff
@@ -100,24 +100,44 @@ class Pdr extends Manager {
     };
 
     record.duration = (record.timestamp - record.createdAt) / 1000;
+    this.constructor.recordIsValid(record, this.schema);
     return record;
   }
 
   /**
-   * Create a new PDR record from incoming SNS messages
+   * Try to update a PDR record from a cloudwatch event.
+   * If the record already exists, only update if the execution is different (re-run case).
    *
-   * @param {Object} payload - SNS message containing the output of a Cumulus Step Function
-   * @returns {Promise<Object>} a PDR record
+   * @param {Object} cumulusMessage - cumulus message object
    */
-  createPdrFromSns(payload) {
-    const pdrObj = get(payload, 'payload.pdr', get(payload, 'meta.pdr'));
```

Comment:
Also, as much as I want to delete this method, don't we need to deprecate it?

Reply:
Out of curiosity: On what basis? I had thought we'd settled on not wanting ourselves to use the models outside of the API package, presumably the same would apply to users.

If it's just a general "don't delete code without deprecating ever" then we already have an issue with all of the `report-*` lambda code we've deleted. 🙂

Reply:
I guess the basis was my understanding of the convo we had, but I wasn't actually in that meeting. Given that technically any exported code could be being used by someone else. But your call.

Do we need to care/worry about no longer falling back to look for a PDR on `meta.pdr`?

Reply:
> generatePdrRecord does not fallback to looking for a PDR on meta.pdr
I don't think we want to check `meta.pdr` and here's why:
If an `ingestGranule` execution has a `meta.PDR` field, then it would trigger a record update. However, there would not be any `stats` values in the payload, so the `progress` value would be set to `0`. Because this is a different execution, an update *would* occur but the progress would be reset to `0`. Presumably the `ParsePdr` execution would finish later and everything would end up correct in the end, but I don't think it's a valuable use case.

Reply:
OK. Aren't we relying on the convention there that the IngestGranule execution will always have the PDR on `meta.pdr` though? That could change, right?

Reply:
Discussed offline and determined that support for `meta.pdr` had already been deprecated, which also addresses the concerns about the IngestGranule workflow